src/Plugin.php

```diff
@@ -63,6 +63,10 @@ public function activate(Composer $composer, IO\IOInterface $io)
         }
         // @codeCoverageIgnoreEnd
 
+        if (!defined('\Composer\Installer\InstallerEvents::POST_DEPENDENCIES_SOLVING')) {
```

Comment:
What is the reason for this condition?

Reply:
It indicates that composer 2 is used (as stated)? :)

Reply:
But for this `PluginInterface::PLUGIN_API_VERSION` was introduced.